pgx-pg-sys/build.rs

```diff
@@ -634,6 +635,92 @@ fn build_shim_for_version(
     Ok(())
 }
 
+fn extra_bindgen_clang_args(pg_config: &PgConfig) -> eyre::Result<Vec<String>> {
+    let mut out = vec![];
+    if std::env::var("CARGO_CFG_TARGET_OS").unwrap_or_default() == "macos" {
+        // On macOS, find the `-isysroot` arg out of the c preprocessor flags,
+        // to handle the case where bindgen uses a libclang isn't provided by
+        // the system.
+        let flags = pg_config.cppflags()?;
+        // In practice this will always be valid UTF-8 because of how the
+        // `pgx-pg-config` crate is implemented, but even if it were not, the
+        // problem won't be with flags we are interested in.
+        let flags = shlex::split(&flags.to_string_lossy()).unwrap_or_default();
+        // Find the `-isysroot` flags -- The rest are `-I` flags that don't seem
+        // to be needed inside the code (and feel likely to cause bindgen to
+        // emit bindings for unrelated libraries)
+        for pair in flags.windows(2) {
```

Comment:
Hm. This may sound silly but are we sure that `-isysroot` will only have one argument? Is it totally incoherent for it to have two?

Reply:
Yep, definitely one. No `-isysroot={path}` version either, clang does not accept that (Well, if you wanna provide an equals sign, it's gotta be as `--sysroot={path}`, but the behavior is different, because ~~of course it is~~ clang uses that one for compatibility with something else, apparently).